### PR TITLE
Add nodeParent property to HTML, Document and DocumentFragment

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
Add parentNode property to Element, Document and DocumentFragment.
spec: [ParentNode](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode)
original issue: [ ParentNode interface isn't implemented in lib.d.ts #4400 ](https://github.com/Microsoft/TypeScript/issues/4400)

### CODE REVIEW

**Comment by YuichiNukiyama** [comment]
I add childElementCoount : number, latter.

**Comment by YuichiNukiyama** [comment]
In the Living standard, parentNode property of the Node interface return a Node. However, it not implemented because of abstract interface. In order to eliminate the error, should I change the parentNode of the return type?
[4.4. Interface Node](https://dom.spec.whatwg.org/#interface-node)

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -12659,6 +12662,13 @@ declare var HTMLPictureElement: {
     new(): HTMLPictureElement;
 }
 
+interface parentNode {
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: use Camel Case. so `ParentNode`

File: baselines/dom.generated.d.ts [hunk 2]
```diff
@@ -2677,6 +2678,7 @@ declare var Document: {
 }
 
 interface DocumentFragment extends Node, NodeSelector {
+    parentNode: parentNode;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: this should be added to the extends clause and not a property. the spec says `DocumentFragment` implements `ParentNode`, and not that it has a `parentNode` property, all `Node`s have a parentNode alaredy. unless i am missing something

File: baselines/dom.generated.d.ts [hunk 3]
```diff
@@ -12659,6 +12662,13 @@ declare var HTMLPictureElement: {
     new(): HTMLPictureElement;
 }
 
+interface parentNode {
+    children: HTMLCollection;
+    firstElementChild : Element;
+    lastElementChild : Element;
+    childElementCount: number;
+}
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: what about querySelector, and querySelctorAll?

File: inputfiles/addedTypes.json [hunk 4]
```diff
@@ -400,5 +400,46 @@
...
         "signatures": ["getElementsByTagName(tagname: \"picture\"): NodeListOf<HTMLPictureElement>"]
+    },
+    {
+        "kind": "interface",
+        "name": "parentNode",
+        "flavor": "DOM",
+        "properties": [
+            {
+                "name": "children",
+                "type": "HTMLCollection"
```
Comment (mhegazy) [inline inputfiles/addedTypes.json]: i would add a "read-only" property here and make it "1". thought we do not support this in the script now, we should soon. this way we get the correct declaration later on. that also applies to the other properties.

**Comment by mhegazy** [comment]
`parentNode: Node` seems fine to me,

File: baselines/dom.generated.d.ts (hunk 1 above)
Comment (YuichiNukiyama) [inline baselines/dom.generated.d.ts]: did it.

File: baselines/dom.generated.d.ts (hunk 2 above)
Comment (YuichiNukiyama) [inline baselines/dom.generated.d.ts]: I see. But, I can't override extending target.
[#50](https://github.com/Microsoft/TSJS-lib-generator/issues/50)

File: baselines/dom.generated.d.ts (hunk 3 above)
Comment (YuichiNukiyama) [inline baselines/dom.generated.d.ts]: HTML, Document and DocumentFragment interface already had been implimented NodeSelector interface. So, I think that no selector methods are required for ParentNode interface.

File: inputfiles/addedTypes.json (hunk 4 above)
Comment (YuichiNukiyama) [inline inputfiles/addedTypes.json]: Should I send New Issue?

Comment (mhegazy) [inline inputfiles/addedTypes.json]: sure.

File: inputfiles/addedTypes.json [hunk 5]
```diff
@@ -400,5 +400,40 @@
...
+                "type": "Element"
+            },
+            {
+                "name": "childElementCount",
+                "type": "number"
+            }
+        ]
+    },
+    {
+        "kind": "property",
```
Comment (mhegazy) [inline inputfiles/addedTypes.json]: i do not think this should be a property.

File: inputfiles/addedTypes.json (hunk 4 above)
Comment (YuichiNukiyama) [inline inputfiles/addedTypes.json]: did it. #67

File: inputfiles/addedTypes.json (hunk 5 above)
Comment (YuichiNukiyama) [inline inputfiles/addedTypes.json]: Is this interface, too?

Comment (mhegazy) [inline inputfiles/addedTypes.json]: it should be added to the extends list. i do not know how to do that though :) @zhengbli what is the right way to do this?

**Comment by mhegazy** [comment]
@zhengbli any ideas?

**Comment by zhengbli** [comment]
The support for adding `extends` is being added at #79 , once that merge in you can update the PR. Thanks!

**Comment by YuichiNukiyama** [comment]
@zhengbli @mhegazy I update addType.json L404 to L438

**Comment by YuichiNukiyama** [comment]
I missed operation. I close this PR and redo.

**Pull request closed without merging**